gcp-endpoint/backend.tf

```diff
@@ -0,0 +1,43 @@
+locals {
+  backend_group = compact([
+    var.cloud_run_service_name != null ? google_compute_region_network_endpoint_group.cloud_run[0].id : null,
+    var.cloud_function_service_name != null ? google_compute_region_network_endpoint_group.cloud_function[0].id : null
+  ])
+  is_serverless = var.cloud_run_service_name != null || var.cloud_function_service_name != null
+  health_checks = local.is_serverless ? null : [google_compute_health_check.mig[0].id]
+}
+
+resource "google_compute_backend_service" "main" {
+  name        = var.resource_name
+  timeout_sec = 30
+
+  dynamic "backend" {
+    for_each = var.managed_instance_groups == null ? {} : var.managed_instance_groups
+    content {
+      group = backend.value.instance_group
+    }
+  }
+
+  dynamic "backend" {
+    for_each = local.is_serverless ? [true] : []
+    content {
+      group = local.backend_group[0]
+    }
+  }
+
+  health_checks = local.health_checks
+}
+
+resource "google_compute_health_check" "mig" {
+  count               = local.is_serverless ? 0 : 1
+  name                = var.resource_name
+  check_interval_sec  = 5
+  timeout_sec         = 5
+  healthy_threshold   = 2
+  unhealthy_threshold = 10 # 50 seconds
+
+  http_health_check {
+    request_path = "/"
```

Comment:
IMHO this is a reasonable default but this should be configurable. So many apps have a `/health` or `/healthz` path explicitly for this purpose.

this matters because often times you can health check the service without auth-ing the request. However, hitting the root path may require auth or redirect for login kinda thing.

Reply:
Great point! I don't know about Elixir but our other example apps don't currently use other paths for healthchecks so I was lazy here. It would be nice to solve this along with startup probes for k8s. How do you feel about making this a variable on the module but burying a default there.

Reply:
+1 to configurable health check paths